From fc77105272f92f1a4783ea9a7b600541204faa65 Mon Sep 17 00:00:00 2001
From: enthropy7 <enthropy7@users.noreply.github.com>
Date: Wed, 26 Aug 2026 03:26:24 +0300
Subject: [PATCH] arm64: dts: allwinner: h5: add the TV encoder

The H5 has the same TV encoder as the H3 at a different address, and
the second mixer and TCON that feed it are already described in the
shared H3/H5 dtsi. Add the encoder node and the resets it needs. It
stays disabled, like the rest of the pipeline.

Build-tested only, I have no H5 board.

Signed-off-by: enthropy7 <enthropy7@users.noreply.github.com>
---
 arch/arm64/boot/dts/allwinner/sun50i-h5.dtsi | 22 ++++++++++++++++++++
 1 file changed, 22 insertions(+)

diff --git a/arch/arm64/boot/dts/allwinner/sun50i-h5.dtsi b/arch/arm64/boot/dts/allwinner/sun50i-h5.dtsi
index c211f2825..afd78ef2a 100644
--- a/arch/arm64/boot/dts/allwinner/sun50i-h5.dtsi
+++ b/arch/arm64/boot/dts/allwinner/sun50i-h5.dtsi
@@ -199,6 +199,20 @@ ths: thermal-sensor@1c25000 {
 			nvmem-cell-names = "calibration";
 			#thermal-sensor-cells = <1>;
 		};
+
+		tve: tv-encoder@1e40000 {
+			compatible = "allwinner,sun50i-h5-tv-encoder";
+			reg = <0x01e40000 0x1000>;
+			clocks = <&ccu CLK_BUS_TVE>;
+			resets = <&ccu RST_BUS_TVE>;
+			status = "disabled";
+
+			port {
+				tve_in_tcon1: endpoint {
+					remote-endpoint = <&tcon1_out_tve>;
+				};
+			};
+		};
 	};
 
 	thermal-zones {
@@ -303,6 +317,10 @@ &mbus {
 	compatible = "allwinner,sun50i-h5-mbus";
 };
 
+&mixer1 {
+	resets = <&display_clocks RST_MIXER1>;
+};
+
 &mmc0 {
 	compatible = "allwinner,sun50i-h5-mmc",
 		     "allwinner,sun50i-a64-mmc";
@@ -338,3 +356,7 @@ &rtc {
 &sid {
 	compatible = "allwinner,sun50i-h5-sid";
 };
+
+&tcon1_out_tve {
+	remote-endpoint = <&tve_in_tcon1>;
+};
